### PR TITLE
Keyvault ARM template for live test

### PR DESCRIPTION
**Pull request by hemanttanwar**
Our goal is to create common Azure resources using ARM template for each module. And pipeline to manage common resources and drop them in end. This will save cost and remove unwanted resources after test is complete or stopped in between

The pipeline will run ARM template and create common resources for example resource group, Accounts etc.
The pipeline will pass on names of these resources as ENV variable into test. For example AZURE_KEYVAULT_ENDPOINT
In the end pipeline will also delete these common resources which it created earlier and that will cause all the resources created to be removed.

### CODE REVIEW

**Comment by hemanttanwar** [comment]
Some of the live test failure are known and Vinay is working on them.

File: sdk/keyvault/test-resources.json [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            "properties": {
+                "sku": {
+                    "family": "A",
+                    "name": "standard"
+                },
+                "tenantId": "[parameters('tenantId')]",
+                "accessPolicies": [
+                    {
+                        "tenantId": "[parameters('tenantId')]",
+                        "objectId": "[parameters('testApplicationOid')]",
```
Comment (g2vinay) [inline sdk/keyvault/test-resources.json]: IS this application object id, of the azure-sdk-tests (service principal)
@hemanttanwar @danieljurek

**Pull request merged**